src/main/java/com/louis/HashSet.java

```diff
@@ -0,0 +1,186 @@
+package com.louis;
+
+import com.google.common.base.Preconditions;
+import java.util.Set;
+import java.util.List;
+import java.util.ArrayList;
+import java.lang.reflect.Array;
+
+/**
+ * Implements a hashset.
+ *
+ * Hashsets implement the set abstract data type. The set contains unique elements. A hash function
+ * computes the location of an element.The location is checked to see if it contains the element.
+ * This implementation merely modifies the inbuilt HashMap.
+ *
+ * @author Louis Ashton (louisashton@live.com)
+ */
+public class HashSet<E> {
+
+    private static final int MIN_CAPACITY = 32;
+    private static final double RESIZE_LOAD = 0.5;
+    private static final double MAX_LOAD = 0.75;
+    private static final double MIN_LOAD = 0.25;
+
+    private HashEntry[] set;
+    private int currentSize;
+    private int currentCapacity;
+
+    /**
+     * Creates a new HashSet.
+     */
+    @SuppressWarnings("unchecked")
+    public HashSet() {
+        currentSize = 0;
+        currentCapacity = MIN_CAPACITY;
+        set = (HashEntry[]) Array.newInstance(HashEntry.class, currentCapacity);
+    }
+
+    /**
+     * Defines a HashTable entry.
+     * 
+     * Each entry consists of a element and an associated value. Both can be retrieved.
+     */
+    private class HashEntry {
+
+        private final E element;
+
+        /**
+         * Creates a new HashEntry.
+         */
+        HashEntry(E element) {
+            this.element = element;
+        }
+
+        /**
+         * Returns the element.
+         * 
+         * @return Returns the entries' element.
+         */
+        public E getElement() {
+            return element;
+        }
+    }
+
+    /**
+     * Checks the set for an element.
+     *
+     * @param element is to be checked.
+     * @return Returns true if the set contains that value.
+     * @throws IllegalArgumentException if element is null.
+     */
+    public synchronized boolean contains(E element) {
+        Preconditions.checkArgument(element != null,
+                "First argument to contains(E element) is null");
+        // Finds the element.
+        int index = hash(element) % currentCapacity;
+        while (!element.equals(set[index].getElement())) {
+            index = (index + 1) % currentCapacity;
+        }
+        return set[index] != null;
+    }
+
+    /**
+     * Adds an element to the set.
+     *
+     * @param element is the element to be added.
```

Comment:
suggest `the element to be added` (don't make the param the first word in the sentence.